### PR TITLE
Remove all occurences of retried tests on success

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Small modification to fast unit test function to avoid some false positives. Remove all occurences of failure when the test succeed after retry

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/test.py [hunk 1]
```diff
@@ -1296,6 +1296,9 @@ def parse_test_log(log_file):
             if json_line["Action"] == "pass" and "Test" in json_line:
                 n_test_executed += 1
                 if f'{json_line["Package"]}/{json_line["Test"]}' in failed_tests:
-                    failed_tests.remove(f'{json_line["Package"]}/{json_line["Test"]}')
-
+                    failed_tests = [
+                        failed_test
+                        for failed_test in failed_tests
+                        if f'{json_line["Package"]}/{json_line["Test"]}' != failed_test
+                    ]
```
Comment (Pythyu) [inline tasks/test.py]: ❓ Question :
More by curiosity, but why `remove` caused flake ? The only difference between what you added and remove is that remove removes the first occurence and here it remove all occurence. But is it possible to have multiple entry of the same `Package/Test` ?

**Review by Pythyu — APPROVED** [review]
Left a comment otherwise LGTM

File: tasks/test.py (hunk 1 above)
Comment (KevinFairise2) [reply tasks/test.py]: Yes the test can be retried more than once. WIth the current implementation if a test is retried three times, fails twice and then succeed, remove will only remove one occurences and the test will be in the failed tests list while it succeeded at the end. I modified the code to rather not have duplicates in the `failed_tests` list

**Pull request merged**